edit/lint.js

```diff
@@ -0,0 +1,253 @@
+      content.parentNode.style.maxHeight = newMaxHeight;
+    }
+  }
+}
+
+function gotoLintIssue(event) {
+  const issue = event.target.closest('tr');
+  if (!issue) {
+    return;
+  }
+  const block = issue.closest('table');
+  makeSectionVisible(block.cm);
+  block.cm.focus();
+  block.cm.setSelection({
+    line: parseInt(issue.querySelector('td[role="line"]').textContent) - 1,
+    ch: parseInt(issue.querySelector('td[role="col"]').textContent) - 1
+  });
+}
+
+function toggleLintReport() {
+  document.getElementById('lint').classList.toggle('collapsed');
+}
+
+function showLintHelp() {
+  let list = '<ul class="rules">';
+  let header = '';
+  if (prefs.get('editor.linter') === 'csslint') {
+    header = t('issuesHelp', '<a href="https://github.com/CSSLint/csslint" target="_blank">CSSLint</a>');
+    list += CSSLint.getRules().map(rule =>
+      '<li><b>' + rule.name + '</b><br>' + rule.desc + '</li>'
+    ).join('');
+  } else {
+    const rules = [];
+    const url = 'https://stylelint.io/user-guide/rules/';
+    header = t('issuesHelp', `<a href="${url}" target="_blank">stylelint</a>`);
+    // to-do: change this to a generator
+    $$('#lint td[role="severity"]').forEach(el => {
+      const rule = el.title.replace('Rule: (', '').replace(/[()]/g, '').trim();
+      if (!rules.includes(rule)) {
+        list += `<li><a target="_blank" href="${url}${rule}/">${rule}</a></li>`;
+        rules.push(rule);
+      }
+    });
+  }
+  return showHelp(t('issues'), header + list + '</ul>');
+}
+
+function setupStylelintSettingsEvents() {
+  let timer;
+  $('#help-popup .save').addEventListener('click', () => {
+    try {
+      setStylelintRules(JSON.parse($('#help-popup textarea').value).rules);
+      // it is possible to have stylelint rules popup open & switch to csslint
+      if (prefs.get('editor.linter') === 'stylelint') {
+        updateLinter('stylelint');
+      }
+    } catch (err) {
+      $('#help-popup .error').classList.add('show');
+      clearTimeout(timer);
+      timer = setTimeout(() => {
+        // popup may be closed at this point
+        const error = $('#help-popup .error');
+        if (error) {
+          error.classList.remove('show');
+        }
+      }, 3000);
+    }
+    return false;
+  });
+  $('#help-popup .reset').addEventListener('click', () => {
+    setStylelintRules();
+    $('#help-popup .settings').value = JSON.stringify({rules: stylelintDefaultConfig.rules}, null, 2);
+    if (prefs.get('editor.linter') === 'stylelint') {
+      updateLinter('stylelint');
+    }
+    return false;
+  });
+}
+
+function openStylelintSettings() {
+  BG.chromeLocal.getValue('editorStylelintRules').then((rules = stylelintDefaultConfig.rules) => {
+    const link = '<a target="_blank" href="https://stylelint.io/demo/">Stylelint</a>';
+    const text = JSON.stringify({rules: rules}, null, 2);
+    const content = `<textarea class="contents settings">${text}</textarea>
+      <p>${t('setStylelintLink', link)}</p>
```

Comment:
Eventually we'll replace all tHTML usage along with the function itself. Meanwhile we use it only for translated messages, and $element for the new DOM-building code.